superset/views/core.py

```diff
@@ -2532,6 +2532,20 @@ def queries(self, last_updated_ms):
         # UTC date time, same that is stored in the DB.
         last_updated_dt = utils.EPOCH + timedelta(seconds=last_updated_ms_int / 1000)
 
+        # Before you run the select query, also update the query.
+
+        now = int(round(time.time() * 1000))
+
+        update(Query).where(Query.user_id == g.user.get_id()).\
```

Comment:
Nope, we can't run an update every time we poll, this runs hundreds of times per second at Airbnb. This won't work.

Reply:
@mistercrunch  Made updated it to only run the query if it definitely has some query in a bad state that's been there for over the timeout period.